include/swift/AST/ArchetypeBuilder.h

```diff
@@ -302,7 +302,9 @@ class ArchetypeBuilder {
   ///
   /// \param allowConcreteGenericParams If true, allow generic parameters to
   /// be made concrete.
-  void finalize(SourceLoc loc, bool allowConcreteGenericParams=false);
+  void finalize(SourceLoc loc,
+                ArrayRef<GenericTypeParamType *> genericParams,
```

Comment:
Why do you need to pass in the generic parameters? Aren't they already available inside the ArchetypeBuilder as `Impl-> GenericParams`?